website/src/pages/admin/index.tsx

```diff
@@ -0,0 +1,67 @@
+import Head from "next/head";
+import { useRouter } from "next/router";
+import { useSession } from "next-auth/react";
+import { useEffect } from "react";
+import { getTransparentHeaderLayout } from "src/components/Layout";
+import UsersCell from "src/components/UsersCell";
+
+/**
+ * Provides the admin index page that will display a list of users and give
+ * admins the ability to manage their access rights.
+ */
+const AdminIndex = () => {
+  const router = useRouter();
+  const { data: session, status } = useSession();
+
+  // Check when the user session is loaded and re-route if the user is not an
+  // admin.  This follows the suggestion by NextJS for handling private pages:
+  //   https://nextjs.org/docs/api-reference/next/router#usage
+  //
+  // All admin pages should use the same check and routing steps.
+  useEffect(() => {
+    if (status === "loading") {
+      return;
+    }
+    if (session?.user?.role === "admin") {
+      return;
+    }
+    router.push("/");
+  }, [session, status]);
+
+  // While loading, just show something.
+  if (status === "loading") {
```

Comment:
could you please extract shared stuff out of the if statement?

the only thing different is contents of `main`:

```jsx
<main className="oa-basic-theme">
{status === "loading": "loading..." : <UserCell/>}
</main>
```

Reply:
Done.  This came from a bit of moving things around and I didn't realize it could be cleaner.